File: 0007/0007.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2021/05/03 13:22
# File: 0007.py
# Desc: 

class Solution:
    def reverse(self, x: int) -> int:
        s = str(x)
        if s.startswith("-"):
            s = '-' + s[1:][::-1]
        else:
            s = s[::-1]
            
        ans = int(s)
        
        if -2147483648 > ans or 2147483647 < ans:
            ans = 0
        
        return ans

File: Solutions/0007/0007.java
/**
 * @Author: washing
 * @DateTime: 2021/01/21 14:01
 * @File: 0007.java
 * @Desc: 
 **/

class Solution {
    public int reverse(int x) {
        
        long n = 0;
        while(x != 0) {
            n = n * 10 + x % 10;
            x = x / 10;
        }
        if (n < -2147483648 || n > 2147483647){
            return 0;
        } else {
            return (int)n;
        }
    }
}